modules/core/arrow-core-data/src/main/kotlin/arrow/core/ListK.kt

```diff
@@ -173,6 +173,65 @@ data class ListK<out A>(private val list: List<A>) : ListKOf<A>, List<A> by list
 
   override fun hashCode(): Int = list.hashCode()
 
+  /**
+   * Align two Lists as in zip, but filling in blanks with None.
+   */
+  fun <B> padZip(
+    other: ListK<B>
+  ): ListK<Tuple2<Option<A>, Option<B>>> =
+    alignWith(this, other) { ior ->
+      ior.bimap({ Option.just(it) }, { Option.just(it) })
+        .fold(
+          { it toT Option.empty<B>() },
+          { Option.empty<A>() toT it },
+          { a, b -> a toT b })
+    }
```

Comment:
This is not quite what I meant.
```kotlin
ior.fold({ it.some() toT None }, { None toT it.some() }, { l, r -> l.some() toT r.some() })
```
Does the same thing and skips the call to bimap. For a small function like this I think this is better, especially if the fold is needed anyway.

Reply:
true, this is more concise